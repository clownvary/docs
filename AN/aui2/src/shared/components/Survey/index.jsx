import React from 'react';
import classNames from 'classnames';
import Survey from 'react-base-ui/lib/components/Survey';
import { count } from 'react-base-ui/lib/utils/dataAccess';
import Collapse from 'react-base-ui/lib/components/Collapse';
import * as helper from './helper';
import { APIQuestionFormat, APIQuestionType } from './consts';

import './index.less';

class AUISurvey extends React.PureComponent {
  addQuestion = () => {
    const { addQuestion, addQuestionList, disabledAddQuestions } = this.props;
    if (!count(addQuestionList) || disabledAddQuestions) {
      return;
    }
    addQuestion();
  }

  render() {
    const {
      questions,
      changeQuestion,
      deleteQuestion,
      isQuestionsChanged,
      showQuestions,
      containerClassName,
      readOnly,
      canDelete,
      addableQuestionsLoaded,
      addQuestionList,
      permitID,
      disabledAddQuestions
    } = this.props;

    let addQuestionsLength = 0;
    const isReservationDetail = permitID > 0;
    if (isReservationDetail && addQuestionList) {
      addQuestionsLength = count(addQuestionList);
    }

    let data = helper.transformQuestions(
      helper.getShownQuestions(questions), canDelete, readOnly);

    if (addableQuestionsLoaded) {
      data = helper.transformAddQuestions(addQuestionList, data, canDelete, readOnly);
    }

    const panelHeader = (
      <span>
        <span>Custom Question</span>
        {isQuestionsChanged && <span className="update-changed-label">UPDATED</span>}
      </span>
    );

    return (
      <div>
        {
          showQuestions &&
            (<div className={classNames('survey-container', containerClassName)}>
              <Collapse activeKey="survey">
                <Collapse.Panel Header={panelHeader} key="survey">
                  <div className="header-section">
                    <span>QUESTION DESCRIPTION</span>
                    {
                      isReservationDetail && !addableQuestionsLoaded &&
                      (
                        <span
                          className={classNames('header-section__add-link', { 'is-disabled': (disabledAddQuestions || !addQuestionsLength) })}
                          onClick={this.addQuestion}
                        >
                          <i className="icon icon-plus-circle" />
                          Add custom question
                        </span>
                      )
                    }
                  </div>
                  <Survey
                    questionLabelClass="afx-xl-3-12"
                    questionContainerClass="afx-xl-5-12"
                    data={data}
                    onChange={({ path, value }) =>
                      changeQuestion({ questionPath: path, answer: value })
                    }
                    onIconClick={({ path }) => deleteQuestion({ questionPath: path })}
                  />
                </Collapse.Panel>
              </Collapse>
            </div>)
        }
      </div>
    );
  }
}

export default AUISurvey;

export {
  AUISurvey,
  helper,
  APIQuestionFormat,
  APIQuestionType
};
