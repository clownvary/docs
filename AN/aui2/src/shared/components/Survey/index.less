@import '~shared/style/variables/index.less';
/* stylelint-disable */

.survey-container {
  .survey-body {
    padding-top: 20px;

    .an-survey {
      .question {
        &-label__name {
          span:last-child {
            word-wrap: break-word;
          }
        }

        &-answer {
          padding-top: 5px;

          .aaui-dropdown,
          .input-group {
            max-width: 310px;
          }

          .radio,
          .checkbox {
            word-break: break-all;
          }

          .an-phone > span {
            line-height: 30px;
          }

          &__error {
            padding: 1px 0 10px 0;
          }
        }

        &-icon {
          width: 33.33333333%;

          .link {
            width: 24px;
            text-align: center;
          }

          .icon {
            font-size: 20px;
          }
        }
      }
    }
  }
}
