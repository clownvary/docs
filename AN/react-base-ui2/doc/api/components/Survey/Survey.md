## Survey 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
data|array|&ensp;|false|An array that content type must be same with Question/QuestionGroup PropTypes, you can check Question/QuestionGroup component doc.
onChange|func|&ensp;|true|The callback function that is triggered when changes the answer value.
onIconClick|func|&ensp;|false|The callback function that is triggered when clicks the post question icon.