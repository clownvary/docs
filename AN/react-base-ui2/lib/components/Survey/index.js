'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.Survey = exports.InputType = exports.QuestionType = exports.QuestionGroup = exports.Question = undefined;var _Question = require('./Question');var _Question2 = _interopRequireDefault(_Question);
var _QuestionGroup = require('./QuestionGroup');var _QuestionGroup2 = _interopRequireDefault(_QuestionGroup);
var _Survey = require('./Survey');var _Survey2 = _interopRequireDefault(_Survey);
var _consts = require('./consts');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

_Survey2.default;exports.
Question = _Question2.default;exports.QuestionGroup = _QuestionGroup2.default;exports.QuestionType = _consts.QuestionType;exports.InputType = _consts.InputType;exports.Survey = _Survey2.default;