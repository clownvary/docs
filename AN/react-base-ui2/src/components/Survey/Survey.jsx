import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Question from './Question';
import QuestionGroup from './QuestionGroup';
import { HORIZONTAL, VERTICAL } from './consts/QuestionLayout';

export const SurveyPropTypes = {
  /**
   *
   */
  layout: PropTypes.oneOf([HORIZONTAL, VERTICAL]),
  /**
   * An array that content type must be same with Question/QuestionGroup PropTypes,
   * you can check Question/QuestionGroup component doc.
   */
  data: PropTypes.array,
  /**
   * The callback function that is triggered when changes the answer value.
   */
  onChange: PropTypes.func.isRequired,
  /**
   * The callback function that is triggered when clicks the post question icon.
   */
  onIconClick: PropTypes.func
};

const SurveyDefaultProps = {
  layout: HORIZONTAL
};

export default class Survey extends React.PureComponent {
  static displayName = 'Survey';
  static propTypes = SurveyPropTypes;
  static defaultProps = SurveyDefaultProps;

  render() {
    const { layout, data, onChange, onIconClick, questionLabelClass,
      questionContainerClass } = this.props;
    return (
      <div
        className={classNames('an-survey', {
          'an-survey__vertical': layout === VERTICAL
        })}
      >
        {
          data.map((item, index) => {
            if (item.groupId) {
              return (<QuestionGroup
                key={`group_${item.groupId}`}
                questions={item.questions}
                onChange={onChange}
                onIconClick={onIconClick}
                groupId={item.groupId}
                groupName={item.groupName}
                isBoldBorder={item.isBoldBorder}
                questionLabelClass={questionLabelClass}
                questionContainerClass={questionContainerClass}
              />);
            } else if (item.type === 'divider') {
              return (
                <div
                  key={index}
                  className={classNames('an-survey-divider', item.className)}
                  style={item.style || {}}
                />);
            }
            return (<Question
              key={item.path || item.id}
              {...item}
              onChange={onChange}
              onIconClick={onIconClick}
              questionLabelClass={questionLabelClass}
              questionContainerClass={questionContainerClass}
            />);
          })
        }
      </div>
    );
  }
}
