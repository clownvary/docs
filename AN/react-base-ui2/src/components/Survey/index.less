@import '../../less/variables/index.less';
/* stylelint-disable */

.an-survey {
  .questiongroup {
    border-bottom: 1px dotted @color-gray-lighter;

    &.is-bold-border {
      border-bottom: 2px solid @color-gray-lighter;
    }

    &-header {
      display: flex;
      padding-top: 15px;
      padding-bottom: 10px;
      font-size: @font-size-base;

      .icon-circle {
        margin: 7px 10px 7px 0;
        font-size: 7px;
        cursor: default;
      }
    }

    &-body .question-label {
      padding-left: 24px;
    }
  }

  .question {
    padding-top: 15px;
    padding-bottom: 10px;
    font-size: 16px;
    border-bottom: 1px dotted @color-gray-lighter;

    &.is-bold-border {
      border-bottom: 2px solid @color-gray-lighter;
    }

    &:last-child {
      border-bottom: none;
    }

    &-label {
      flex-wrap: nowrap;
      padding-right: 5px;
      vertical-align: top;

      &__name {
        display: flex;
        padding-bottom: 10px;

        .icon-asterisk,
        .icon-circle {
          margin: 7px 10px 7px 0;
          font-size: 7px;
          cursor: default;
        }

        .icon-asterisk {
          color: red;
        }
      }

      &__hint {
        padding-left: 18px;
        font-size: 14px;
        color: @color-hint;
      }
    }

    &-answer {
      .radio-wrapper,
      .checkbox-wrapper {
        min-height: 20px;
        padding: 5px 0;

        .radio,
        .checkbox {
          display: flex;

          span {
            display: inline-block;
            flex: 1;
          }

          span.radio__inner {
            flex: none;
            margin-top: 2px;
          }

          span.checkbox__inner {
            flex: none;
            margin-top: 1px;
          }
        }
      }

      &__error {
        font-size: 16px;
        color: red;

        &:before {
          font-size: 20px;
        }
      }
    }

    .question-icon {
      margin-top: 5px;
      text-align: right;
      vertical-align: top;
      color: @color-blue;

      .link {
        cursor: pointer;

        &.is-disabled {
          cursor: not-allowed;
          color: @color-hint;
          opacity: 0.45;
        }
      }
    }

    .afx-col {
      font-size: 16px;
    }
  }

  .question-error {
    .question-label__name {
      color: red;
    }

    .input-group {
      &__field {
        color: @state-danger-text;
        background-color: @state-danger-bg;
        border: solid 1px @state-danger-border !important;
      }
    }

    .aaui-dropdown {
      border-radius: 3px;

      button {
        background-color: @state-danger-bg;
        border: solid 1px @state-danger-border;
      }
    }
  }

  .checkbox-group {
    display: flex;
    flex-flow: column;

    .checkbox-wrapper {
      min-height: 30px;
      padding: 0 0 10px;
    }
  }

  .radio-group {
    display: flex;
    flex-flow: column;

    .radio-wrapper {
      min-height: 30px;
      padding: 0 0 10px;
    }
  }

  &-divider {
    display: block;
    width: 100%;
    margin-bottom: 10px;
    border-bottom: 1px solid @color-gray-lighter;
  }
}
