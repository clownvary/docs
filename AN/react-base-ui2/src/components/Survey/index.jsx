import Question from './Question';
import QuestionGroup from './QuestionGroup';
import Survey from './Survey';
import { QuestionType, InputType } from './consts';

export default Survey;
export { Question, QuestionGroup, QuestionType, InputType, Survey };
