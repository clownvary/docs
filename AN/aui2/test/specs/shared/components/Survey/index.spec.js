import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import Survey from 'shared/components/Survey';
import baseSurvey from 'react-base-ui/lib/components/Survey';

describe('shared -> components -> Survey', () => {
  const setup = (overrideProps) => {
    const defaultProps = {
      questions: [{
        permitQuestion: true,
        parentQuestionIndex: 0,
        readOnly: false,
        path: [0],
        question: 'numbersOnly',
        hint: '',
        isMainQuestion: true,
        parentQuestionID: 0,
        answerMaxLength: 100,
        customquestionID: 524,
        useAnswerCode: false,
        answers: [],
        answerFormat: 'numbersOnly',
        answerType: 'userEntry',
        title: 'numbersOnly',
        customquestionIndex: '184',
        answerRequired: false,
        defaultAnswer: ''
      }],
      readOnly: false,
      changeQuestion: jest.fn(),
      deleteQuestion: jest.fn(),
      isQuestionsChanged: false,
      showQuestions: true,
      containerClassName: '',
      canDelete: true
    };

    const props = { ...defaultProps, ...overrideProps };

    const component = mount(<Survey {...props} />);

    return {
      component,
      props
    };
  };
  const addQuestion = jest.fn();

  const addQuestionList = [{
    title: 'CoreyDropdown',
    question: 'CoreyDropdown',
    hint: '',
    path:[0],
    answers: [{
      answer: '#1',
      code: 0,
      selected: true,
      disabled: false,
      answerID: 1222,
      subQuestion: {
        title: 'CoreyQuestion_UserEntry_PhoneNumber',
        question: 'CoreyQuestion_UserEntry_PhoneNumber',
        hint: '',
        path:[0],
        answers: [],
        customquestionIndex: '(3699_0)_2_1',
        customquestionID: 315,
        useAnswerCode: false,
        answerType: 'userEntry',
        answerRequired: true,
        answerFormat: 'phoneNumber',
        answerMaxLength: 100,
        defaultAnswer: '1111111111',
        postalCodeType: 0,
        answer: '111-1111111-',
        permitQuestion: false,
        readOnly: false
      }
    }, {
      answer: '#2',
      code: 0,
      selected: false,
      disabled: false,
      answerID: 1223,
      subQuestion: null
    }],
    customquestionIndex: '(3699_0)_2',
    customquestionID: 324,
    useAnswerCode: false,
    answerType: 'singleSelectionDropdown',
    answerRequired: false,
    answerFormat: 'freeForm',
    answerMaxLength: 100,
    defaultAnswer: '',
    postalCodeType: 0,
    answer: null,
    permitQuestion: false,
    readOnly: false
  }, {
    questions: [{
      title: 'CoreyQuestion_SingleSelection_Dropdown_2',
      question: 'CoreyQuestion_SingleSelection_Dropdown',
      hint: '',
      path:[0],
      answers: [{
        answer: '1',
        code: 0,
        selected: true,
        disabled: false,
        answerID: 1209,
        subQuestion: {
          title: 'CoreyQuestion_UserEntry_Freeform',
          question: 'CoreyQuestion_UserEntry_Freeform',
          hint: '',
          path:[0],
          answers: [],
          customquestionIndex: '(3699_0)_3_1',
          customquestionID: 321,
          useAnswerCode: false,
          answerType: 'userEntry',
          answerRequired: true,
          answerFormat: 'freeForm',
          answerMaxLength: 100,
          defaultAnswer: 'Free form default answer',
          postalCodeType: 0,
          answer: 'Free form default answer',
          permitQuestion: false,
          readOnly: false
        }
      }, {
        answer: '2',
        code: 0,
        selected: false,
        disabled: false,
        answerID: 1210,
        subQuestion: {
          title: 'CoreyQuestion_MultiSelection-Listbox',
          question: 'CoreyQuestion_MultiSelection-Listbox',
          hint: '',
          path:[0],
          answers: [{
            answer: '1',
            code: 0,
            selected: false,
            disabled: false,
            answerID: 1178,
            subQuestion: null
          }, {
            answer: '2',
            code: 0,
            selected: false,
            disabled: false,
            answerID: 1179,
            subQuestion: null
          }, {
            answer: '3',
            code: 0,
            selected: false,
            disabled: false,
            answerID: 1180,
            subQuestion: null
          }],
          customquestionIndex: '(3699_0)_3_2',
          customquestionID: 312,
          useAnswerCode: false,
          answerType: 'multiSelectionListbox',
          answerRequired: true,
          answerFormat: 'freeForm',
          answerMaxLength: 100,
          defaultAnswer: '',
          postalCodeType: 0,
          answer: null,
          permitQuestion: false,
          readOnly: false
        }
      }],
      customquestionIndex: '(3699_0)_3',
      customquestionID: 317,
      useAnswerCode: false,
      answerType: 'singleSelectionDropdown',
      answerRequired: false,
      answerFormat: 'freeForm',
      answerMaxLength: 100,
      defaultAnswer: '',
      postalCodeType: 0,
      answer: null,
      permitQuestion: false,
      readOnly: false
    }],
    groupID: 26,
    groupOrder: 1,
    groupHeader: 'CoreyGroup',
    customquestionIndex: null,
    postalCodeType: 0,
    answer: null
  }]

  it('render component correctly', () => {
    const { component } = setup();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('render component correctly when questions is empty', () => {
    const { component } = setup({ questions: [] });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('render component correctly when showQuestions is false', () => {
    const { component } = setup({ showQuestions: false });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('render component correctly when isQuestionsChanged is true', () => {
    const { component } = setup({ isQuestionsChanged: true });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('question section can be shown or hidden correctly', () => {
    const { component, props } = setup({ isQuestionsChanged: true });

    expect(component.find(baseSurvey).length).toEqual(1);
    component.find(baseSurvey).props().onChange({ path: [0], value: '1' });
    expect(props.changeQuestion).toBeCalled();

    component.find(baseSurvey).props().onIconClick({ path: [0] });
    expect(props.deleteQuestion).toBeCalled();
  });

  it('when permitID > 0 and addQuestionsList > 0 render component correctly', () => {
    const { component } = setup({
      addableQuestionsLoaded: false,
      permitID: 23,
      questions: [...addQuestionList],
      addQuestionList,
      addQuestion
    });
    component.find('.header-section__add-link').simulate('click');
    expect(addQuestion).toBeCalled();

    component.setProps({ addableQuestionsLoaded: true });
  });

  it('when permitID > 0 and addQuestionsLength = 0 render component correctly', () => {
    const { component } = setup({
      addableQuestionsLoaded: false,
      permitID: 23,
      addQuestionList: [],
      addQuestion
    });
    const addList = component.find('.header-section__add-link');
    addList.simulate('click');
    expect(addList.hasClass('is-disabled')).toBe(true);
  });
});

