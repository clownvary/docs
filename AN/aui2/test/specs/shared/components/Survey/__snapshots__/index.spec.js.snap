// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared -> components -> Survey render component correctly 1`] = `
<AUISurvey
  canDelete={true}
  changeQuestion={[Function]}
  containerClassName=""
  deleteQuestion={[Function]}
  isQuestionsChanged={false}
  questions={
    Array [
      Object {
        "answerFormat": "numbersOnly",
        "answerMaxLength": 100,
        "answerRequired": false,
        "answerType": "userEntry",
        "answers": Array [],
        "customquestionID": 524,
        "customquestionIndex": "184",
        "defaultAnswer": "",
        "hint": "",
        "isMainQuestion": true,
        "parentQuestionID": 0,
        "parentQuestionIndex": 0,
        "path": Array [
          0,
        ],
        "permitQuestion": true,
        "question": "numbersOnly",
        "readOnly": false,
        "title": "numbersOnly",
        "useAnswerCode": false,
      },
    ]
  }
  readOnly={false}
  showQuestions={true}
>
  <div>
    <div
      className="survey-container"
    >
      <Collapse
        activeKey="survey"
        dataSource={Array []}
        multiple={true}
        onChange={[Function]}
        prefixCls="an-collapse"
      >
        <div
          className="an-collapse"
        >
          <Panel
            Header={
              <span>
                <span>
                  Custom Question
                </span>
              </span>
            }
            headerButtonId="default"
            isActive={true}
            key="survey"
            onItemClick={[Function]}
            prefixCls="an-collapse"
          >
            <div
              className="an-collapse-item is-expanded"
            >
              <a
                aria-expanded={true}
                className="an-collapse-item__header"
                href="javascript:void(0)"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  onClick={[Function]}
                >
                  <span>
                    <span>
                      Custom Question
                    </span>
                  </span>
                  <i
                    className="icon icon-chevron-down"
                    onClick={[Function]}
                    tabIndex={0}
                  />
                </div>
              </a>
              <div
                className="an-collapse-item__content"
              >
                <div
                  className="header-section"
                >
                  <span>
                    QUESTION DESCRIPTION
                  </span>
                </div>
                <Survey
                  data={
                    Array [
                      Object {
                        "answers": Array [],
                        "errorMsgs": Array [],
                        "format": 8,
                        "hint": "",
                        "icon": "icon-trash",
                        "id": "184",
                        "isShown": true,
                        "maxLength": 100,
                        "onChange": [Function],
                        "onIconClick": [Function],
                        "path": Array [
                          0,
                        ],
                        "placeHolder": "",
                        "readonly": false,
                        "required": false,
                        "text": "numbersOnly",
                        "type": 0,
                        "value": "",
                      },
                    ]
                  }
                  layout="horizontal"
                  onChange={[Function]}
                  onIconClick={[Function]}
                  questionContainerClass="afx-xl-5-12"
                  questionLabelClass="afx-xl-3-12"
                >
                  <div
                    className="an-survey"
                  >
                    <Question
                      answers={Array []}
                      errorMsgs={Array []}
                      format={8}
                      hint=""
                      icon="icon-trash"
                      id="184"
                      isShown={true}
                      key="0"
                      maxLength={100}
                      onChange={[Function]}
                      onIconClick={[Function]}
                      path={
                        Array [
                          0,
                        ]
                      }
                      placeHolder=""
                      questionContainerClass="afx-xl-5-12"
                      questionLabelClass="afx-xl-3-12"
                      readonly={false}
                      required={false}
                      text="numbersOnly"
                      type={0}
                      value=""
                    >
                      <div
                        className="question aaui-flex"
                        key="question_184"
                      >
                        <div
                          className="afx-col question-label afx-xl-3-12"
                        >
                          <div
                            className="question-label__name"
                          >
                            <span
                              className="icon icon-circle"
                            />
                            <SafeText
                              className=""
                              dangerMode={true}
                              data-qa-id=""
                              decode={true}
                              tagName="span"
                              text="numbersOnly"
                            >
                              <span
                                className=""
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "numbersOnly",
                                  }
                                }
                                data-qa-id=""
                              />
                            </SafeText>
                          </div>
                          <div
                            className="question-label__hint"
                          >
                            <SafeText
                              dangerMode={true}
                              data-qa-id=""
                              decode={true}
                              tagName="span"
                              text=""
                            >
                              <span
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "",
                                  }
                                }
                                data-qa-id=""
                              />
                            </SafeText>
                          </div>
                        </div>
                        <div
                          className="afx-col question-answer-container afx-xl-5-12"
                        >
                          <div
                            className="question-answer"
                          >
                            <Input
                              disabled={false}
                              formula={/\\\\d/}
                              icon={false}
                              inputRef={[Function]}
                              maxLength={100}
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder=""
                              size="md"
                              type="text"
                              value=""
                            >
                              <div
                                className="input-group input-group--md"
                              >
                                <input
                                  className="input input-group__field input__field"
                                  disabled={false}
                                  maxLength={100}
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onInput={[Function]}
                                  onKeyPress={[Function]}
                                  onPaste={[Function]}
                                  placeholder=""
                                  type="text"
                                  value=""
                                />
                              </div>
                            </Input>
                          </div>
                        </div>
                        <div
                          className="afx-col afx-xl-1-12 question-icon"
                        >
                          <span
                            className="icon icon-trash link"
                            onClick={[Function]}
                          />
                        </div>
                      </div>
                    </Question>
                  </div>
                </Survey>
              </div>
            </div>
          </Panel>
        </div>
      </Collapse>
    </div>
  </div>
</AUISurvey>
`;

exports[`shared -> components -> Survey render component correctly when isQuestionsChanged is true 1`] = `
<AUISurvey
  canDelete={true}
  changeQuestion={[Function]}
  containerClassName=""
  deleteQuestion={[Function]}
  isQuestionsChanged={true}
  questions={
    Array [
      Object {
        "answerFormat": "numbersOnly",
        "answerMaxLength": 100,
        "answerRequired": false,
        "answerType": "userEntry",
        "answers": Array [],
        "customquestionID": 524,
        "customquestionIndex": "184",
        "defaultAnswer": "",
        "hint": "",
        "isMainQuestion": true,
        "parentQuestionID": 0,
        "parentQuestionIndex": 0,
        "path": Array [
          0,
        ],
        "permitQuestion": true,
        "question": "numbersOnly",
        "readOnly": false,
        "title": "numbersOnly",
        "useAnswerCode": false,
      },
    ]
  }
  readOnly={false}
  showQuestions={true}
>
  <div>
    <div
      className="survey-container"
    >
      <Collapse
        activeKey="survey"
        dataSource={Array []}
        multiple={true}
        onChange={[Function]}
        prefixCls="an-collapse"
      >
        <div
          className="an-collapse"
        >
          <Panel
            Header={
              <span>
                <span>
                  Custom Question
                </span>
                <span
                  className="update-changed-label"
                >
                  UPDATED
                </span>
              </span>
            }
            headerButtonId="default"
            isActive={true}
            key="survey"
            onItemClick={[Function]}
            prefixCls="an-collapse"
          >
            <div
              className="an-collapse-item is-expanded"
            >
              <a
                aria-expanded={true}
                className="an-collapse-item__header"
                href="javascript:void(0)"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  onClick={[Function]}
                >
                  <span>
                    <span>
                      Custom Question
                    </span>
                    <span
                      className="update-changed-label"
                    >
                      UPDATED
                    </span>
                  </span>
                  <i
                    className="icon icon-chevron-down"
                    onClick={[Function]}
                    tabIndex={0}
                  />
                </div>
              </a>
              <div
                className="an-collapse-item__content"
              >
                <div
                  className="header-section"
                >
                  <span>
                    QUESTION DESCRIPTION
                  </span>
                </div>
                <Survey
                  data={
                    Array [
                      Object {
                        "answers": Array [],
                        "errorMsgs": Array [],
                        "format": 8,
                        "hint": "",
                        "icon": "icon-trash",
                        "id": "184",
                        "isShown": true,
                        "maxLength": 100,
                        "onChange": [Function],
                        "onIconClick": [Function],
                        "path": Array [
                          0,
                        ],
                        "placeHolder": "",
                        "readonly": false,
                        "required": false,
                        "text": "numbersOnly",
                        "type": 0,
                        "value": "",
                      },
                    ]
                  }
                  layout="horizontal"
                  onChange={[Function]}
                  onIconClick={[Function]}
                  questionContainerClass="afx-xl-5-12"
                  questionLabelClass="afx-xl-3-12"
                >
                  <div
                    className="an-survey"
                  >
                    <Question
                      answers={Array []}
                      errorMsgs={Array []}
                      format={8}
                      hint=""
                      icon="icon-trash"
                      id="184"
                      isShown={true}
                      key="0"
                      maxLength={100}
                      onChange={[Function]}
                      onIconClick={[Function]}
                      path={
                        Array [
                          0,
                        ]
                      }
                      placeHolder=""
                      questionContainerClass="afx-xl-5-12"
                      questionLabelClass="afx-xl-3-12"
                      readonly={false}
                      required={false}
                      text="numbersOnly"
                      type={0}
                      value=""
                    >
                      <div
                        className="question aaui-flex"
                        key="question_184"
                      >
                        <div
                          className="afx-col question-label afx-xl-3-12"
                        >
                          <div
                            className="question-label__name"
                          >
                            <span
                              className="icon icon-circle"
                            />
                            <SafeText
                              className=""
                              dangerMode={true}
                              data-qa-id=""
                              decode={true}
                              tagName="span"
                              text="numbersOnly"
                            >
                              <span
                                className=""
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "numbersOnly",
                                  }
                                }
                                data-qa-id=""
                              />
                            </SafeText>
                          </div>
                          <div
                            className="question-label__hint"
                          >
                            <SafeText
                              dangerMode={true}
                              data-qa-id=""
                              decode={true}
                              tagName="span"
                              text=""
                            >
                              <span
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "",
                                  }
                                }
                                data-qa-id=""
                              />
                            </SafeText>
                          </div>
                        </div>
                        <div
                          className="afx-col question-answer-container afx-xl-5-12"
                        >
                          <div
                            className="question-answer"
                          >
                            <Input
                              disabled={false}
                              formula={/\\\\d/}
                              icon={false}
                              inputRef={[Function]}
                              maxLength={100}
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder=""
                              size="md"
                              type="text"
                              value=""
                            >
                              <div
                                className="input-group input-group--md"
                              >
                                <input
                                  className="input input-group__field input__field"
                                  disabled={false}
                                  maxLength={100}
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onInput={[Function]}
                                  onKeyPress={[Function]}
                                  onPaste={[Function]}
                                  placeholder=""
                                  type="text"
                                  value=""
                                />
                              </div>
                            </Input>
                          </div>
                        </div>
                        <div
                          className="afx-col afx-xl-1-12 question-icon"
                        >
                          <span
                            className="icon icon-trash link"
                            onClick={[Function]}
                          />
                        </div>
                      </div>
                    </Question>
                  </div>
                </Survey>
              </div>
            </div>
          </Panel>
        </div>
      </Collapse>
    </div>
  </div>
</AUISurvey>
`;

exports[`shared -> components -> Survey render component correctly when questions is empty 1`] = `
<AUISurvey
  canDelete={true}
  changeQuestion={[Function]}
  containerClassName=""
  deleteQuestion={[Function]}
  isQuestionsChanged={false}
  questions={Array []}
  readOnly={false}
  showQuestions={true}
>
  <div>
    <div
      className="survey-container"
    >
      <Collapse
        activeKey="survey"
        dataSource={Array []}
        multiple={true}
        onChange={[Function]}
        prefixCls="an-collapse"
      >
        <div
          className="an-collapse"
        >
          <Panel
            Header={
              <span>
                <span>
                  Custom Question
                </span>
              </span>
            }
            headerButtonId="default"
            isActive={true}
            key="survey"
            onItemClick={[Function]}
            prefixCls="an-collapse"
          >
            <div
              className="an-collapse-item is-expanded"
            >
              <a
                aria-expanded={true}
                className="an-collapse-item__header"
                href="javascript:void(0)"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  onClick={[Function]}
                >
                  <span>
                    <span>
                      Custom Question
                    </span>
                  </span>
                  <i
                    className="icon icon-chevron-down"
                    onClick={[Function]}
                    tabIndex={0}
                  />
                </div>
              </a>
              <div
                className="an-collapse-item__content"
              >
                <div
                  className="header-section"
                >
                  <span>
                    QUESTION DESCRIPTION
                  </span>
                </div>
                <Survey
                  data={Array []}
                  layout="horizontal"
                  onChange={[Function]}
                  onIconClick={[Function]}
                  questionContainerClass="afx-xl-5-12"
                  questionLabelClass="afx-xl-3-12"
                >
                  <div
                    className="an-survey"
                  />
                </Survey>
              </div>
            </div>
          </Panel>
        </div>
      </Collapse>
    </div>
  </div>
</AUISurvey>
`;

exports[`shared -> components -> Survey render component correctly when showQuestions is false 1`] = `
<AUISurvey
  canDelete={true}
  changeQuestion={[Function]}
  containerClassName=""
  deleteQuestion={[Function]}
  isQuestionsChanged={false}
  questions={
    Array [
      Object {
        "answerFormat": "numbersOnly",
        "answerMaxLength": 100,
        "answerRequired": false,
        "answerType": "userEntry",
        "answers": Array [],
        "customquestionID": 524,
        "customquestionIndex": "184",
        "defaultAnswer": "",
        "hint": "",
        "isMainQuestion": true,
        "parentQuestionID": 0,
        "parentQuestionIndex": 0,
        "path": Array [
          0,
        ],
        "permitQuestion": true,
        "question": "numbersOnly",
        "readOnly": false,
        "title": "numbersOnly",
        "useAnswerCode": false,
      },
    ]
  }
  readOnly={false}
  showQuestions={false}
>
  <div />
</AUISurvey>
`;
